nifi-commons/nifi-record/src/main/java/org/apache/nifi/serialization/record/util/DataTypeUtils.java

```diff
@@ -712,6 +746,20 @@ public static Object[] toArray(final Object value, final String fieldName, final
             return dest;
         }
 
+        if (value instanceof UUID) {
+            UUID uuid = (UUID)value;
+            ByteBuffer buffer = ByteBuffer.allocate(16);
+            buffer.putLong(uuid.getMostSignificantBits());
+            buffer.putLong(uuid.getLeastSignificantBits());
+            Byte[] result = new Byte[16];
+            byte[] array = buffer.array();
+            for (int index = 0; index < array.length; index++) {
+                result[index] = array[index];
+            }
+
+            return result;
```

Comment:
What is the reason for returning an object `Byte[]` type versus a primitive `byte[]` type? It makes sense to support both types on the `toUUID()` method, but not following why the representation here is `Byte[]`.

Reply:
I noticed that everything, even the "to byte array" conversion goes to the boxed byte type, so I followed the convention on the assumption that somewhere there might be an expectation of that method not returning a primitive array.